### PR TITLE
Add test for uploading the same source package twice.

### PR DESCRIPTION
**Pull request by hstct**
[noissue]

### CODE REVIEW

**Review by daviddavis — APPROVED** [review]

File: pulp_deb/tests/functional/api/test_source_package.py [hunk 1]
```diff
@@ -121,3 +121,31 @@ def test_upload_source_package_and_publish(
...
+
+
+def test_upload_same_source_package(
+    artifact_factory,
+    apt_source_package_api,
+    deb_source_package_factory,
+    delete_orphans_pre,
+):
+    """Test whether uploading the same source package works and that it stays unique."""
+    artifact_factory(SOURCE_PACKAGE_SOURCE)
```
Comment (mdellweg) [inline pulp_deb/tests/functional/api/test_source_package.py:133]: Are we using this?

File: pulp_deb/tests/functional/api/test_source_package.py [hunk 2]
```diff
@@ -121,3 +121,31 @@ def test_upload_source_package_and_publish(
     # attempt to publish the repo
     publication = deb_publication_factory(repository, structured=True)
     assert publication.repository_version == repository.latest_version_href
+
+
+def test_upload_same_source_package(
+    artifact_factory,
+    apt_source_package_api,
+    deb_source_package_factory,
+    delete_orphans_pre,
```
Comment (mdellweg) [inline pulp_deb/tests/functional/api/test_source_package.py:130]: If you created the package into a repository, there'd be no need for orphan cleanup and the test should be able to run with "@pytest.mark.parallel".

**Review by mdellweg — COMMENTED** [review]
Fell free to ignore my comments and merge anyway.

File: pulp_deb/tests/functional/api/test_source_package.py (hunk 1 above)
Comment (daviddavis) [reply pulp_deb/tests/functional/api/test_source_package.py:133]: Yes, the way it works is that you upload the source(s) and then the source control file (dsc) which has the digests of the sources. The multi-artifact source package then gets created from the dsc and any source artifacts it points to.

File: pulp_deb/tests/functional/api/test_source_package.py (hunk 2 above)
Comment (hstct) [reply pulp_deb/tests/functional/api/test_source_package.py:130]: Huh, I might try to do that!

Comment (hstct) [reply pulp_deb/tests/functional/api/test_source_package.py:130]: It's actually more trouble than it's worth. The testing time is still reasonable once it gets out of hand again one can look at making more tests run in parallel again.

**Pull request merged**